feat(Insomnia): Konnect integration (#5288)
Kong/developer.konghq.com: app/insomnia/konnect-integration.md

  - q: Can I use the {{ site.konnect_short_name }} integration with my on-prem Gateway?
    a: |
      No, the {{ site.konnect_short_name }} integration only works with Gateway Services hosted in {{ site.konnect_short_name }}, not on-prem {{site.base_gateway}}.

---

## About the {{ site.konnect_short_name }} + {{ site.data.products.insomnia.name }} integration

Starting from {{ site.data.products.insomnia.name }} `v13`, Enterprise users can link {{ site.data.products.insomnia.name }} to Gateway Services deployed in {{ site.konnect_short_name }}. The integration allows sending requests from Collections against routes pulled from {{ site.konnect_short_name }}, using the {{ site.data.products.insomnia.name }} app.

After connecting and syncing your Gateway Services, {{ site.data.products.insomnia.name }}:

- Displays a list of your Gateway Services from {{ site.konnect_short_name }}.
- Pulls the routes from each Gateway Service.
- Lists the routes in Collections for each Gateway Service.

## Prerequisites

- An {{ site.data.products.insomnia.name }} [Enterprise account](/insomnia/enterprise/)
- A {{ site.konnect_short_name }} [Gateway Service](/gateway/)

> 

Reviewer: Happy to have this as a FAQ, pre-req or whatever you think, but I think we should call out that authenticating requires a Konnect PAT right now, and that in the next few months we'll extend the integration to Dev Portal and Catalog so that PATs (and Konnect access) are not necessary.

Author: It's here, next to the authenticate feature: https://deploy-preview-5288--kongdeveloper.netlify.app/insomnia/konnect-integration/#available-features

    usage: "Link {{ site.data.products.insomnia.name }} to your Gateway Service"
    how: |
      1. From {{ site.konnect_short_name }}, generate a [Personal Access Token (PAT)](/konnect-api/#personal-access-tokens).
      1. Copy the PAT.
      1. In Insomnia, open **Preferences** > **Konnect**.
      1. Paste the PAT.
      1. Click **Validate & Save**.
  - feature: |
      **Konnect** tab
    usage: |
      List your gateway services in {{ site.konnect_short_name }}, next to the **Projects** tab
    how: |
      Open the personal workspace linked to {{ site.konnect_short_name }}.
  - feature: "Sync"
    usage: "Pull the latest updates from {{ site.konnect_short_name }} into {{ site.data.products.insomnia.name }}."
    how: |
      From the **Konnect** tab, click **Sync**.
{% endtable %}

### Syncing

Every time you click **Sync** from the {{ site.konnect_short_name }} tab, {{ site.data.products.insomnia.name }} pulls changes from {{ site.konnect_short_name }}. Syncing:

- Doesn't push any changes from {{ site.data.products.insomnia.name }} to {{ site.konnect_short_name }}.
- Pulls any changes or configurations from {{ site.konnect_short_name }} for the related routes.

> Syncing preserves or resets the following data for each pulled route in {{ site.data.products.insomnia.name }}:

Reviewer: I think we can be clearer here that we preserve their local changes so they can keep working, while also updating as often as they like with the source of truth.

Author: ```suggestion

This preserves the local changes on routes while allowing pulling any updates from {{ site.konnect_short_name }}.

Syncing preserves or resets the following data for each pulled route in {{ site.data.products.insomnia.name }}:
```

---
title: "{{ site.konnect_short_name }} Gateway Service integration in {{ site.data.products.insomnia.name }}"

> description: "Connect your Gateway Services to test routes in {{ site.data.products.insomnia.name }}." 

Author: ```suggestion
description: "Link {{ site.konnect_short_name }} to send requests from Insomnia against Routes in  your Gateway Services from {{ site.data.products.insomnia.name }}." 
```

content_type: reference
layout: reference

breadcrumbs:
  - /insomnia/
  - /insomnia/enterprise/
min_version:
  insomnia: '13'
products:
    - insomnia
tier: enterprise

next_steps:
  - text: Link a {{ site.konnect_short_name }} Service Gateway to {{ site.data.products.insomnia.name }}
    url: /how-to/link-konnect-to-insomnia/

related_resources:
  - text: Security at Insomnia
    url: /insomnia/manage-insomnia/#security
  - text: Enterprise
    url: /insomnia/enterprise/
  - text: Enterprise account management
    url: /insomnia/enterprise-account-management/
  - text: Enterprise user management
    url: /insomnia/enterprise-user-management/
  - text: Migrate from scratch pad to Enterprise
    url: /insomnia/migrate-from-scratch-pad-to-enterprise/
  - text: Link Konnect to Insomnia
    url: /how-to/link-konnect-to-insomnia/

faqs:

>   - q: Why does {{ site.data.products.insomnia.name }} skip some routes?

Author: ```suggestion
  - q: Why does {{ site.data.products.insomnia.name }} skip some Routes?
```

>   - q: Why does {{ site.data.products.insomnia.name }} skip some Routes?

Reviewer 2: What does skip mean here? Overwriting?

Author: It means you can't test them on Insomnia, they don't work in the app. Maybe we should add this question precisely to the faq.

    a: |

>       {{ site.data.products.insomnia.name }} skips routes that use a protocol Insomnia doesn't support, such as SNI matching, TCP, or UDP. {{ site.data.products.insomnia.name }} displays the skipped routes in a separate collection named "Skipped routes".

Author: ```suggestion
      {{ site.data.products.insomnia.name }} skips Routes that use a protocol {{ site.data.products.insomnia.name }}  doesn't support, such as SNI matching, TCP, or UDP. {{ site.data.products.insomnia.name }} displays the skipped routes in a separate collection named "Skipped routes".
```

>       {{ site.data.products.insomnia.name }} skips Routes that use a protocol {{ site.data.products.insomnia.name }}  doesn't support, such as SNI matching, TCP, or UDP. {{ site.data.products.insomnia.name }} displays the skipped routes in a separate collection named "Skipped routes".

Reviewer 2: ```suggestion
      {{ site.data.products.insomnia.name }} skips Routes that use an unsupported protocol, such as SNI matching, TCP, or UDP. {{ site.data.products.insomnia.name }} displays the skipped routes in a separate collection named "Skipped routes".
```

Ah, I see they are called "skipped routes"

  - q: Can I use the {{ site.konnect_short_name }} integration with my on-prem Gateway?
    a: |
      No, the {{ site.konnect_short_name }} integration only works with Gateway Services hosted in {{ site.konnect_short_name }}, not on-prem {{site.base_gateway}}.

---

> ## About the {{ site.konnect_short_name }} + {{ site.data.products.insomnia.name }} integration

Reviewer 2: ```suggestion
## About the {{ site.konnect_short_name }} integration
```

I think Insomnia might be implied here.

> Starting from {{ site.data.products.insomnia.name }} `v13`, Enterprise users can link {{ site.data.products.insomnia.name }} to Gateway Services deployed in {{ site.konnect_short_name }}. The integration allows sending requests from Collections against routes pulled from {{ site.konnect_short_name }}, using the {{ site.data.products.insomnia.name }} app.

Author: ```suggestion
Starting from {{ site.data.products.insomnia.name }} `v13`, Enterprise users can link {{ site.data.products.insomnia.name }} to Gateway Services deployed in {{ site.konnect_short_name }}. The integration allows sending requests from Collections against Routes pulled from {{ site.konnect_short_name }}, using the {{ site.data.products.insomnia.name }} app.
```

After connecting and syncing your Gateway Services, {{ site.data.products.insomnia.name }}:

- Displays a list of your Gateway Services from {{ site.konnect_short_name }}.

> - Pulls the routes from each Gateway Service.

Author: ```suggestion
- Pulls the Routes from each Gateway Service.
```

> - Lists the routes in Collections for each Gateway Service.

Author: ```suggestion
- Lists the Routes in Collections for each Gateway Service.
```

## Prerequisites

- An {{ site.data.products.insomnia.name }} [Enterprise account](/insomnia/enterprise/)
- A {{ site.konnect_short_name }} [Gateway Service](/gateway/).
- A [Personal Access Token (PAT)](/konnect-api/#personal-access-tokens) 

## Available features

> Collections work the same as for local routes. The following table lists the features that allow you to link {{ site.data.products.insomnia.name }} to {{ site.konnect_short_name }}:

Author: ```suggestion
Collections work the same as for local Routes. The following table lists the features that allow you to link {{ site.data.products.insomnia.name }} to {{ site.konnect_short_name }}:
```

  - title: Usage
    key: usage
  - title: How to
    key: how
rows:
  - feature: "Authenticate"
    usage: "Link {{ site.data.products.insomnia.name }} to your Gateway Service"
    how: |
      1. From {{ site.konnect_short_name }}, generate a [Personal Access Token (PAT)](/konnect-api/#personal-access-tokens).
      1. Copy the PAT.
      1. In Insomnia, open **Preferences** > **Konnect**.
      1. Paste the PAT.
      1. Click **Validate & Save**.
  - feature: |
      **Konnect** tab
    usage: |
      List your gateway services in {{ site.konnect_short_name }}, next to the **Projects** tab
    how: |
      Open the personal workspace linked to {{ site.konnect_short_name }}.
  - feature: "Sync"
    usage: "Pull the latest updates from {{ site.konnect_short_name }} into {{ site.data.products.insomnia.name }}."
    how: |
      From the **Konnect** tab, click **Sync**.
{% endtable %}

### Syncing

Every time you click **Sync** from the {{ site.konnect_short_name }} tab, {{ site.data.products.insomnia.name }} pulls changes from {{ site.konnect_short_name }}. Syncing:

- Doesn't push any changes from {{ site.data.products.insomnia.name }} to {{ site.konnect_short_name }}.

> - Pulls any changes or configurations from {{ site.konnect_short_name }} for the related routes.

Author: ```suggestion
- Pulls any changes or configurations from {{ site.konnect_short_name }} for the related Routes.
```

> This preserves the local changes on routes while allowing pulling any updates from {{ site.konnect_short_name }}.

Author: ```suggestion
This preserves the local changes on Routes while allowing pulling any updates from {{ site.konnect_short_name }}.
```

  - q: Can I use the {{ site.konnect_short_name }} integration with my on-prem Gateway?
    a: |
      No, the {{ site.konnect_short_name }} integration only works with Gateway Services hosted in {{ site.konnect_short_name }}, not on-prem {{site.base_gateway}}.

---

## About the {{ site.konnect_short_name }} + {{ site.data.products.insomnia.name }} integration

Starting from {{ site.data.products.insomnia.name }} `v13`, Enterprise users can link {{ site.data.products.insomnia.name }} to Gateway Services deployed in {{ site.konnect_short_name }}. The integration allows sending requests from Collections against routes pulled from {{ site.konnect_short_name }}, using the {{ site.data.products.insomnia.name }} app.

After connecting and syncing your Gateway Services, {{ site.data.products.insomnia.name }}:

- Displays a list of your Gateway Services from {{ site.konnect_short_name }}.
- Pulls the routes from each Gateway Service.
- Lists the routes in Collections for each Gateway Service.

## Prerequisites

- An {{ site.data.products.insomnia.name }} [Enterprise account](/insomnia/enterprise/)

> - A {{ site.konnect_short_name }} [Gateway Service](/gateway/)

Author: ```suggestion
- A {{ site.konnect_short_name }} [Gateway Service](/gateway/).
- A [Personal Access Token (PAT)](/konnect-api/#personal-access-tokens) 
```

      1. Copy the PAT.
      1. In Insomnia, open **Preferences** > **Konnect**.
      1. Paste the PAT.
      1. Click **Validate & Save**.
  - feature: |
      **Konnect** tab
    usage: |
      List your gateway services in {{ site.konnect_short_name }}, next to the **Projects** tab
    how: |
      Open the personal workspace linked to {{ site.konnect_short_name }}.
  - feature: "Sync"
    usage: "Pull the latest updates from {{ site.konnect_short_name }} into {{ site.data.products.insomnia.name }}."
    how: |
      From the **Konnect** tab, click **Sync**.
{% endtable %}

### Syncing

Every time you click **Sync** from the {{ site.konnect_short_name }} tab, {{ site.data.products.insomnia.name }} pulls changes from {{ site.konnect_short_name }}. Syncing:

- Doesn't push any changes from {{ site.data.products.insomnia.name }} to {{ site.konnect_short_name }}.
- Pulls any changes or configurations from {{ site.konnect_short_name }} for the related routes.

Syncing preserves or resets the following data for each pulled route in {{ site.data.products.insomnia.name }}:

> {% table %}

Reviewer 2: Keep/reset feel ambiguous to me, and with only one item in the reset column it throws off the table visually for me. I think you can get away with just using prose here. 

```
Syncing preserves the following data for each pulled route:

Parameters
Auth
Body
Custom headers
Scripts
Environment variables
{:.info}
Note: Overridden headers are not preserved — they revert to the values configured in Konnect.
```

If you want to keep the table, consider changing the titles from "Keep/Reset", to something like "Preserved locally/ Overwritten by KOnnect"

Author: ```suggestion

```

columns:
  - title: Keep
    key: keep
  - title: Reset
    key: reset
rows:
  - keep: |
      - Parameters
      - Auth
      - Body
      - Custom headers
      - Scripts
      - Environment variables
    reset: |
      - Overridden headers    

> {% endtable %}

Author: ```suggestion
- Parameters
- Auth
- Body
- Custom headers
- Scripts
- Environment variables

{:.info}
Note: Overridden headers aren't preserved — they revert to the values configured in {{ site.konnect_short_name }}.
```

---

> title: "{{ site.konnect_short_name }} Gateway Service integration in {{ site.data.products.insomnia.name }}"

Reviewer 3: I'm wondering if we should drop the `Gateway Service` from the naming?

The integration pulls resources from whatever Konnect Organization is linked to the PAT provided by a user. The hierarchy of Konnect resources pulled then are 

Organization -> Control Planes -> Gateway Services -> Routes

@chayglass84 any thoughts?

Author: I agree, I changed it.

content_type: reference
layout: reference

breadcrumbs:
  - /insomnia/
  - /insomnia/enterprise/
min_version:
  insomnia: '13'
products:
    - insomnia
tier: enterprise

next_steps:
  - text: Link a {{ site.konnect_short_name }} Service Gateway to {{ site.data.products.insomnia.name }}
    url: /how-to/link-konnect-to-insomnia/

related_resources:
  - text: Security at Insomnia
    url: /insomnia/manage-insomnia/#security
  - text: Enterprise
    url: /insomnia/enterprise/
  - text: Enterprise account management
    url: /insomnia/enterprise-account-management/
  - text: Enterprise user management
    url: /insomnia/enterprise-user-management/
  - text: Migrate from scratch pad to Enterprise
    url: /insomnia/migrate-from-scratch-pad-to-enterprise/
  - text: Link Konnect to Insomnia
    url: /how-to/link-konnect-to-insomnia/

faqs:
  - q: Can I use the {{ site.konnect_short_name }} integration with my on-prem Gateway?
    a: |
      No, the {{ site.konnect_short_name }} integration only works with Gateway Services hosted in {{ site.konnect_short_name }}, not on-prem {{site.base_gateway}}.
  - q: What is the "Skipped routes" Collection?
    a: |

>       The "Skipped Routes" Collection contains Routes using an unsupported protocal. You can't use the Collection runner to send requests against them.    

Reviewer 3: ```suggestion
      The "Skipped Routes" Collection contains Routes using an unsupported protocol. You can't use the Collection runner to send requests against them.    
```

    a: |
      {{ site.data.products.insomnia.name }} skips Routes that use an unsupported protocol, such as SNI matching, TCP, or UDP. {{ site.data.products.insomnia.name }} displays the skipped routes in a separate collection named "Skipped routes".
  - q: Can I use the {{ site.konnect_short_name }} integration with my on-prem Gateway?
    a: |
      No, the {{ site.konnect_short_name }} integration only works with Gateway Services hosted in {{ site.konnect_short_name }}, not on-prem {{site.base_gateway}}.

---

## About the {{ site.konnect_short_name }} integration

Starting from {{ site.data.products.insomnia.name }} `v13`, Enterprise users can link {{ site.data.products.insomnia.name }} to Gateway Services deployed in {{ site.konnect_short_name }}. The integration allows sending requests from Collections against Routes pulled from {{ site.konnect_short_name }}, using the {{ site.data.products.insomnia.name }} app.

After connecting and syncing your Gateway Services, {{ site.data.products.insomnia.name }}:

- Displays a list of your Gateway Services from {{ site.konnect_short_name }}.
- Pulls the Routes from each Gateway Service.

> - Lists the Routes in Collections for each Gateway Service.

Reviewer 3: If we're talking post-sync here, we'd display a project per control plane found in Konnect.
Under each project, there is a a collection for each gateway service found for the control plane in Konnect.
In each collection, there is a request for each route found for the gateway service in Konnect.

Author: solved in e0c636281

## Prerequisites

> - An {{ site.data.products.insomnia.name }} [Enterprise account](/insomnia/enterprise/)

Reviewer 3: This won't be available to Enterprise trials, not sure if that is worth calling out @chayglass84

Author: applied in 2e8e0ea2a

---
title: "{{ site.konnect_short_name }} Gateway Service integration in {{ site.data.products.insomnia.name }}"

> description: "Link {{ site.konnect_short_name }} to send requests from Insomnia against Routes in  your Gateway Services from {{ site.data.products.insomnia.name }}." 

Reviewer 2: ```suggestion
description: "Link {{ site.konnect_short_name }} to send requests from {{ site.data.products.insomnia.name }} to Routes in your Gateway Services.
```

Reviewer 2: extra space

Reviewer 2: and double Insomnia

content_type: reference
layout: reference

breadcrumbs:
  - /insomnia/
  - /insomnia/enterprise/
min_version:
  insomnia: '13'
products:
    - insomnia
tier: enterprise

next_steps:

>   - text: Link a {{ site.konnect_short_name }} Service Gateway to {{ site.data.products.insomnia.name }}

Reviewer 2: ```suggestion
  - text: Link a {{ site.konnect_short_name }} Gateway Service to {{ site.data.products.insomnia.name }}
```

    url: /how-to/link-konnect-to-insomnia/

related_resources:
  - text: Security at Insomnia
    url: /insomnia/manage-insomnia/#security
  - text: Enterprise
    url: /insomnia/enterprise/
  - text: Enterprise account management
    url: /insomnia/enterprise-account-management/
  - text: Enterprise user management
    url: /insomnia/enterprise-user-management/
  - text: Migrate from scratch pad to Enterprise
    url: /insomnia/migrate-from-scratch-pad-to-enterprise/
  - text: Link Konnect to Insomnia
    url: /how-to/link-konnect-to-insomnia/

faqs:
  - q: Can I use the {{ site.konnect_short_name }} integration with my on-prem Gateway?
    a: |
      No, the {{ site.konnect_short_name }} integration only works with Gateway Services hosted in {{ site.konnect_short_name }}, not on-prem {{site.base_gateway}}.

>   - q: What is the "Skipped routes" Collection?

Reviewer 2: ```suggestion
  - q: What is the "Skipped Routes" Collection?
```
Consistent with everything else in the doc.

    a: |
      The "Skipped Routes" Collection contains Routes using an unsupported protocol. You can't use the Collection runner to send requests against them.    
  - q: Why does {{ site.data.products.insomnia.name }} skip some Routes?
    a: |

>       {{ site.data.products.insomnia.name }} skips Routes that use an unsupported protocol, such as SNI matching, TCP, or UDP. {{ site.data.products.insomnia.name }} displays the skipped routes in a separate collection named "Skipped routes".

Reviewer 2: ```suggestion
      {{ site.data.products.insomnia.name }} skips Routes that use an unsupported protocol, such as SNI matching, TCP, or UDP. {{ site.data.products.insomnia.name }} displays the skipped Routes in a separate collection named "Skipped Routes".
```

  - q: Can I use the {{ site.konnect_short_name }} integration with my on-prem Gateway?
    a: |

>       No, the {{ site.konnect_short_name }} integration only works with Gateway Services hosted in {{ site.konnect_short_name }}, not on-prem {{site.base_gateway}}.

Reviewer 2: ```suggestion
```
Duplicate

---

## About the {{ site.konnect_short_name }} integration

> Starting from {{ site.data.products.insomnia.name }} `v13`, Enterprise users can link {{ site.data.products.insomnia.name }} to Gateway Services deployed in {{ site.konnect_short_name }}. The integration allows sending requests from Collections against Routes pulled from {{ site.konnect_short_name }}, using the {{ site.data.products.insomnia.name }} app.

Reviewer 2: ```suggestion
Starting from {{ site.data.products.insomnia.name }} 13, Enterprise users can link {{ site.data.products.insomnia.name }} to Gateway Services deployed in {{ site.konnect_short_name }}. The integration allows sending requests from Collections against Routes pulled from {{ site.konnect_short_name }}, using the {{ site.data.products.insomnia.name }} app.
```

After connecting and syncing your Gateway Services, {{ site.data.products.insomnia.name }}:

- Displays the list of your control planes in {{ site.konnect_short_name }} as projects.
- Nests the Gateway Services on each control plane as Collections under each project.
- Pulls the Routes from each Gateway Service.
- Lists the Routes in Collections for each Gateway Service.

> ## Prerequisites

Reviewer 2: ```suggestion
### Requirements
```

> - An {{ site.data.products.insomnia.name }} [Enterprise account](/insomnia/enterprise/) (not trial)

Reviewer 2: ```suggestion
- An {{ site.data.products.insomnia.name }} [Enterprise account](/insomnia/enterprise/)
```

## Available features

Collections work the same as for local Routes. The following table lists the features that allow you to link {{ site.data.products.insomnia.name }} to {{ site.konnect_short_name }}:

{% table %}
columns:
  - title: Feature
    key: feature
  - title: Usage
    key: usage
  - title: How to
    key: how
rows:
  - feature: "Authenticate"
    usage: "Link {{ site.data.products.insomnia.name }} to your Gateway Service"
    how: |
      1. From {{ site.konnect_short_name }}, generate a [Personal Access Token (PAT)](/konnect-api/#personal-access-tokens).
      1. Copy the PAT.
      1. In Insomnia, open **Preferences** > **Konnect**.
      1. Paste the PAT.
      1. Click **Validate & Save**.
  - feature: |
      **Konnect** tab
    usage: |
      List your gateway services in {{ site.konnect_short_name }}, next to the **Projects** tab
    how: |
      Open the personal workspace linked to {{ site.konnect_short_name }}.
  - feature: "Sync"
    usage: "Pull the latest updates from {{ site.konnect_short_name }} into {{ site.data.products.insomnia.name }}."
    how: |
      From the **Konnect** tab, click **Sync**.
{% endtable %}

> ### Syncing

Reviewer 2: ```suggestion
### Sync
```

Every time you click **Sync** from the {{ site.konnect_short_name }} tab, {{ site.data.products.insomnia.name }} pulls changes from {{ site.konnect_short_name }}. Syncing:

- Doesn't push any changes from {{ site.data.products.insomnia.name }} to {{ site.konnect_short_name }}.

> - Pulls any changes or configurations from {{ site.konnect_short_name }} for the related Routes.

Reviewer 2: ```suggestion
- Pulls changes or configurations from {{ site.konnect_short_name }} for the related Routes.
```

---
title: "{{ site.konnect_short_name }} Gateway Service routes integration in {{ site.data.products.insomnia.name }}"

description: Connect your Gateway Service routes and test them in {{ site.data.products.insomnia.name }}. 

content_type: reference
layout: reference

breadcrumbs:
  - /insomnia/
  - /insomnia/enterprise/

> 

Reviewer 4: ```suggestion
min_version:
  insomnia: '13'
```
This should work I think?
